src/module/chat-message/listeners/cards.ts

```diff
@@ -322,14 +322,14 @@ class ChatCards {
      * This allows for damage to be scaled by a multiplier to account for healing, critical hits, or resistance
      */
     static async #rollActorSaves({ event, button, actor, item }: RollActorSavesParams): Promise<void> {
-        if (canvas.tokens.controlled.length > 0) {
+        if (canvas.tokens?.controlled.length > 0) {
             const saveType = button.dataset.save;
             if (!tupleHasValue(SAVE_TYPES, saveType)) {
                 throw ErrorPF2e(`"${saveType}" is not a recognized save type`);
             }
 
             const dc = Number(button.dataset.dc ?? "NaN");
-            for (const token of canvas.tokens.controlled) {
+            for (const token of canvas.tokens?.controlled) {
```

Comment:
I think this needs `?? []` at the end, otherwise it'll still throw because `for (const token of undefined)` would throw

Reply:
Yea, probably